### PR TITLE
Put namespace argument first

### PR DESCRIPTION
**Pull request by johnkerl**
A survey of recent manual-layer methods shows that `execute_sql_query` and `delayed_sql` had the `namespace` argument last, not first as the rest do.

### CODE REVIEW

File: R/manual_layer_delayed.R
Comment (ihnorton) [inline R/manual_layer_delayed.R:58]: `namespace` is entered twice here

**Review by ihnorton — APPROVED** [review]
One nit, approving with the following for your consideration:

I don't _love_ having the namespace first (although I realize I missed this in #69, at least) -- if for no other reason than that the `namespace` feels like line-noise when reading the vignette. There's also still a _slight_ inconsistency in that `compute` takes `namespace` as the second argument. Your call though.

File: R/manual_layer_delayed.R
Comment (johnkerl) [reply R/manual_layer_delayed.R:58]: oof

**Review by Shelnutt2 — COMMENTED** [review]
Is namespace optional? Does it have a client side default like in python?

**Comment by johnkerl** [comment]
> There's also still a slight inconsistency in that compute takes namespace as the second argument

Good point. Issue is that arguments in R can be referred to with/without keywords (as in Python). If namespace were first then `compute(foo)` would be `compute(namespace=foo)` which would not be OK ...

> I don't love having the namespace first

@ihnorton nor I honestly! I was hoping for (mostly) uniformity and it seemed like most things had namespace first, so that was the nearest target to move to.

I wouldn't mind placing namespace always _last_ for that matter.

Perhaps @Shelnutt2 and/or @aaronwolen and/or @eddelbuettel has some thoughts here -- ?

(P.S. let's not worry about backward compatibility -- none of this is in a tagged release yet -- now is the best time to go ahead & straighten all this out.)

**Comment by johnkerl** [comment]
> Is namespace optional? Does it have a client side default like in python?

@Shelnutt2 not until https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/86 -- #86 was (so it happens) accepted & merged before this PR & this PR is not yet rebased atop it.

File: R/manual_layer_delayed.R
Comment (johnkerl) [reply R/manual_layer_delayed.R]: fixed

**Comment by eddelbuettel** [comment]
Without seeing the full picture here I worry a little about two aspects:
- my (possibly wrong) guestimate is that `namespace` might a per-call constant, not unlike `ctx` or a database handle object; if that is the case why schlepp it around for each call, and why give up the most prominent position (the first) which may make it tiring in interactive use
- the first argument can be used as a 'implicit one', for example the piping functions do that: we have an example in the TileDB R package where were pipe along sevaral invocations of `tiledb_query_*` functions all of which take a query object as first argument (and return it silently); my uninformed suspicion is that `Node` may be a candidate for this too

But as I hinted, I may not have the full picture here.

**Comment by johnkerl** [comment]
Thanks @eddelbuettel ! I do (believe I) know from @Shelnutt2 that we do want to move toward supporting different nodes being invoked in different namespaces ...

**Comment by eddelbuettel** [comment]
Sure. Maybe make the `namespace` (it belongs to) an attribute of the `Node`.  Or maybe that doesn't work for other reasons.  

You asked, I answered.

**Comment by johnkerl** [comment]
> Sure. Maybe make the namespace (it belongs to) an attribute of the Node. Or maybe that doesn't work for other reasons.

@eddelbuettel I like this quite a bit actually!

**Comment by johnkerl** [comment]
@eddelbuettel https://docs.tiledb.com/cloud/api-reference/task-graphs#selecting-who-to-charge

**Comment by johnkerl** [comment]
Merging as-is, with the understanding that on a subsequent PR we'll need to (a) move namespace further down -- consistently -- in the argument lists, and (b) store these in `Node` objects (as factoried by `delayed*` functions) to support the case when people want to charge different nodes to different  namespaces.

**Comment by johnkerl** [comment]
@Shelnutt2 @ihnorton @eddelbuettel https://github.com/TileDB-Inc/TileDB-Cloud-R/pull/88

**Pull request merged**
